src/query/pools/query_pools.go

```diff
@@ -53,7 +53,7 @@ func BuildWorkerPools(
 	cfg config.Configuration,
 	logger *zap.Logger,
 	scope tally.Scope,
-) (pool.ObjectPool, instrument.Options) {
+) (pool.ObjectPool, xsync.WorkerPool, instrument.Options) {
```

Comment:
Out of curiousity, why is there a pool of worker pools?

Reply:
Addressed this quesiton in the other comment